src/solidity/decodeInfo.js

```diff
@@ -0,0 +1,353 @@
+}
+
+/**
+  * Address decode the given @arg type
+  *
+  * @param {String} type - type given by the AST
+  * @return {Object} returns decoded info about the current type: { needsFreeStorageSlot, storageBytes, typeName, decoder}
+  */
+function Address (type) {
+  return {
+    needsFreeStorageSlot: false,
+    storageBytes: 20,
+    typeName: 'address',
+    name: 'address'
+  }
+}
+
+/**
+  * Bool decode the given @arg type
+  *
+  * @param {String} type - type given by the AST
+  * @return {Object} returns decoded info about the current type: { needsFreeStorageSlot, storageBytes, typeName, decoder}
+  */
+function Bool (type) {
+  return {
+    needsFreeStorageSlot: false,
+    storageBytes: 1,
+    typeName: 'bool',
+    name: 'bool'
+  }
+}
+
+/**
+  * DynamicByteArray decode the given @arg type
+  *
+  * @param {String} type - type given by the AST
+  * @return {Object} returns decoded info about the current type: { needsFreeStorageSlot, storageBytes, typeName, decoder}
+  */
+function DynamicByteArray (type) {
+  return {
+    needsFreeStorageSlot: true,
+    storageBytes: 32,
+    typeName: 'bytes',
+    name: 'dynamicByteArray'
+  }
+}
+
+/**
+  * FixedByteArray decode the given @arg type
+  *
+  * @param {String} type - type given by the AST
+  * @return {Object} returns decoded info about the current type: { needsFreeStorageSlot, storageBytes, typeName, decoder}
+  */
+function FixedByteArray (type) {
+  if (type.split(' ')) {
+    type = type.split(' ')[0]
+  }
+  return {
+    needsFreeStorageSlot: false,
+    storageBytes: parseInt(type.replace('bytes', '')),
+    typeName: type.split(' ')[0],
+    name: 'fixedByteArray'
+  }
+}
+
+/**
+  * Int decode the given @arg type
+  *
+  * @param {String} type - type given by the AST
+  * @return {Object} returns decoded info about the current type: { needsFreeStorageSlot, storageBytes, typeName, decoder}
+  */
+function Int (type) {
+  if (type.split(' ')) {
+    type = type.split(' ')[0]
+  }
+  return {
+    needsFreeStorageSlot: false,
+    storageBytes: parseInt(type.replace('int', '')) / 8,
+    typeName: type,
+    name: 'int'
+  }
+}
+
+/**
+  * StringType decode the given @arg type
+  *
+  * @param {String} type - type given by the AST
+  * @return {Object} returns decoded info about the current type: { needsFreeStorageSlot, storageBytes, typeName, decoder}
+  */
+function StringType (type) {
+  return {
+    needsFreeStorageSlot: true,
+    storageBytes: 32,
+    typeName: 'string',
+    name: 'stringType'
+  }
+}
+
+/**
+  * ArrayType decode the given @arg type
+  *
+  * @param {String} type - type given by the AST
+  * @return {Object} returns decoded info about the current type: { needsFreeStorageSlot, storageBytes, typeName, decoder, arraySize, subArray}
+  */
+function ArrayType (type, stateDefinitions) {
+  var arraySize
+  var storageBytes
+
+  var underlyingType = extractUnderlyingType(type)
+
+  arraySize = extractArraySize(type)
+  var dimensions = extractDimensions(type)
```

Comment:
Please ignore dimensions. if we have a multi-dimesional array, the underlying type is just again an array.